cradle-cassandra/src/main/java/com/exactpro/cradle/cassandra/ReadThroughBookCache.java

```diff
@@ -0,0 +1,130 @@
+        this.schemaVersion = schemaVersion;
+    }
+
+    public BookInfo getBook (BookId bookId) {
+        if (!books.containsKey(bookId)) {
+            logger.info("Could not find book named {} in cache, tyring getting from db", bookId.getName());
+            try {
+                books.put(bookId, loadBook(bookId));
+            } catch (Exception e) {
+                logger.warn("Could not find book named {} in database", bookId.getName());
+            }
+        }
+
+        return books.get(bookId);
+    }
+
+    public Collection<PageInfo> loadPageInfo(BookId bookId) throws IOException
+    {
+        Collection<PageInfo> result = new ArrayList<>();
+        try
+        {
+            for (PageEntity pageEntity : ops.getOperators(bookId, readAttrs).getPageOperator().getAll(bookId.getName(), readAttrs))
+            {
+                if (pageEntity.getRemoved() == null)
+                    result.add(pageEntity.toPageInfo());
+            }
+        }
+        catch (Exception e)
+        {
+            throw new IOException("Error while loading pages of book '"+bookId+"'", e);
+        }
+        return result;
+    }
+
+    public BookInfo loadBook (BookId bookId) throws IOException {
+        try {
+            BookEntity bookEntity = ops.getCradleBookOperator().get(bookId.getName(), readAttrs);
+
+            if (!bookEntity.getSchemaVersion().equals(schemaVersion)) {
+                throw new IOException(String.format("Unsupported schema version for the book \"%s\". Expected: %s, found: %s. Skipping",
+                        bookEntity.getName(),
+                        schemaVersion,
+                        bookEntity.getSchemaVersion()));
+            }
+
+            return processBookEntity(bookEntity);
+        } catch (Exception e) {
+            throw new IOException("Error while loading book", e);
+        }
+    }
+
+    private BookInfo processBookEntity (BookEntity entity) throws IOException {
+        try {
+            BookId bookId = new BookId(entity.getName());
+            ops.addOperators(bookId, entity.getKeyspaceName());
+            Collection<PageInfo> pages = loadPageInfo(bookId);
+
+            return entity.toBookInfo(pages);
+        } catch (Exception e) {
+            throw new IOException(String.format("Error while loading book \"%s\"", entity.getName()), e);
+        }
+    }
+
+    public Collection<BookInfo> loadBooks() throws IOException
+    {
+        Collection<BookInfo> result = new ArrayList<>();
+        try
+        {
+            for (BookEntity bookEntity : ops.getCradleBookOperator().getAll(readAttrs))
+            {
+                if(!bookEntity.getSchemaVersion().equals(schemaVersion)) {
+                    logger.warn("Unsupported schema version for the book \"{}\". Expected: {}, found: {}. Skipping",
+                            bookEntity.getName(),
+                            schemaVersion,
+                            bookEntity.getSchemaVersion()
```

Comment:
This message is copied from `loadBook`. Better use `String.format` with constant pattern or a function, returning this message.

Reply:
Resolved, [8d758ea](https://github.com/th2-net/cradleapi/pull/85/commits/8d758ea710b501ede06ef3cf1cb5bb4a1699c435)